Macrophages/Macrophages_multiome.py

```diff
@@ -0,0 +1,746 @@
+                html.Div([
+                    html.H3("Co-regulated molecules"),
+
+                    # Message Div
+                    html.Div(
+                        id='metabolite-message',
+                        children="",  # Initially empty
+                        style={'color': 'red', 'marginBottom': '10px'}
+                    ),
+
+                    # DataTable
+                    dash_table.DataTable(
+                        id='metabolite-data-table',
+                        columns=[],  # Columns will be populated by callback
+                        data=[],  # Data will be populated by callback
+                        page_size=15,  # Display 15 rows per page
+                        style_table={'overflowX': 'auto'},
+                        style_cell={
+                            'textAlign': 'left',
+                            'padding': '5px'
+                        },
+                        style_header={
+                            'backgroundColor': 'rgb(230, 230, 230)',
+                            'fontWeight': 'bold'
+                        },
+                        sort_action='native',  # Enable sorting
+                        filter_action='native'  # Enable filtering
+                    )
+                ], style={'width': '38%', 'display': 'inline-block', 'paddingLeft': '2%'})
+            ], style={'display': 'flex', 'flexDirection': 'row', 'justifyContent': 'space-between'}),
+
+            # Second Row: Heatmap spanning full width
+            html.Div([
+                dcc.Graph(
+                    id='metabolite-heatmap',
+                    config={'displayModeBar': False},  # Optional: Hide the mode bar
+                    style={'marginTop': '40px'}  # Add top margin for spacing
+                )
+            ], style={'width': '100%', 'paddingTop': '20px'}),
+
+            # Third Row: Network Plots spanning full width
+            html.Div([
+                # First Network Plot
+                html.Div([
+                    dcc.Graph(
+                        id='metabolite-network-plot-1',
+                        config={'displayModeBar': False},
+                        style={'marginTop': '40px'}
+                    )
+                ], style={'width': '45%', 'display': 'inline-block'}),
+
+                # Second Network Plot
+                html.Div([
+                    dcc.Graph(
+                        id='metabolite-network-plot-2',
+                        config={'displayModeBar': False},
+                        style={'marginTop': '40px'}
+                    )
+                ], style={'width': '40%', 'display': 'inline-block', 'paddingLeft': '5%'})
+            ], style={'width': '100%', 'paddingTop': '20px'})
+        ])
+
+
+# Callback to update the protein scatter-box plot
+@app.callback(
+    Output('scatter-box-plot-protein', 'figure'),
+    [Input('protein-dropdown', 'value')]
+)
+def update_scatterbox_pro_plot(protein_name):

```

Comment:
Might as well just turn 'pro' -> 'protein' in this function name, since 'pro' can have multiple meanings.